subprojects/internal-testing/src/main/groovy/org/gradle/testing/internal/util/RetryRule.groovy

```diff
@@ -54,6 +54,9 @@ class RetryRule implements MethodRule {
 
     @Override
     Statement apply(Statement base, FrameworkMethod method, Object target) {
+        if (specification == null) {
+            specification = target as Specification
```

Comment:
I think we should even consider removing the specification from the constructor. It could be an argument to the closure. I didn't want to make that change since I am not too familiar with the code here.